src/Microsoft.OData.Core/ODataResource.cs

```diff
@@ -176,6 +176,15 @@ public IEnumerable<ODataFunction> Functions
             get { return this.MetadataBuilder.GetFunctions(); }
         }
 
+        /// <summary>
+        /// Gets or sets whether to skip property verification. When set to false (default behaviour),
+        /// the properties collection will be verified to ensure it doesn't contain invalid values.
+        /// When set to true, it's the caller's responsibility to ensure the property values are valid
+        /// before setting the <see cref="Properties"/> property. This can be a useful optimization
+        /// in hot paths when you're sure property values are valid.
+        /// </summary>
+        public bool SkipPropertyVerification { get; set; }

```

Comment:
Wouldn't this be better as a global setting (e.g. `ODataMessageWriterSettings`/`ODataMessageReaderSettings`) instead of a property that has to be set on each `ODataResource` instance?

Reply:
Problem is `ODataResource` doesn't have access to the message writer settings.

Reply:
Maybe the serializer that creates ODataResources could use the `ODataMessageWritter` settings to decide whether to set the skip verification flag, but in that case it would still have to set it for every resource.

Reply:
"it would still have to set it for every resource" - Why would it need to be set for every resource?

A different question, by which means did you envision that the `SkipPropertyVerification` would be set?

Reply:
I agree with @gathogojr 
Can we use the same setup as the `IDuplicatePropertyNameChecker` where the serializer have access to the `ODataMessageWriteSettings` and this set in the `WriterValidator`

Reply:
@gathogojr 

    I don't think it's any of the ODataResource methods that calls methods for verifying or writing the properties. It's the serializer that does so... If

In the existing code, it is actually the `ODataResource` that calls the method for verifying properties, it happens inside the `ODataResourceBase.Properties` setter. That's why I applied the flag to the `ODataResource`.

Each `ODataResource` instance is independent, and the `ODataMessageWriter` operates on a resource-per-resource basis. The `ODataMessageWriter` does not know about the serializer.

@KenitoInc does have access to the the `ODataMessageWriterSettings` but the `ODataResource` does not. Currently the verification happens within the `ODataResource` itself, not the serializer nor the message writer.

We could change things up such that the verification is moved out of the `ODataResource` and moved into the message writer. Or we could remove the verification altogether and let the serializer handle that manually if it deems fit. I'm not sure if this delayed verification would have unintended consequences in those where verification is desired.